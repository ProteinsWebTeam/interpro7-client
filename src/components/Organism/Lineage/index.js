import React, { PureComponent } from 'react';
import T from 'prop-types';
import { createSelector } from 'reselect';

import Link from 'components/generic/Link';

import loadData from 'higherOrder/loadData';
import description2path from 'utils/processLocation/description2path';

class LineageItem extends PureComponent {
  static propTypes = {
    taxId: T.string.isRequired,
    data: T.object.isRequired,
  };

  render() {
    const { taxId, data } = this.props;
    const displayedText =
      (data.payload &&
        data.payload.metadata &&
        data.payload.metadata.scientific_name) ||
      taxId;
    const newTo = {
      description: {
        mainType: 'organism',
        mainDB: 'taxonomy',
        mainAccession: taxId.toString(),
      },
    };
    return (
      <span>
        {' '}â†’ <Link newTo={newTo}>{displayedText}</Link>
      </span>
    );
  }
}

const getUrlFor = taxId =>
  createSelector(
    state => state.settings.api,
    ({ protocol, hostname, port, root }) =>
      `${protocol}//${hostname}:${port}${root}${description2path({
        mainType: 'organism',
        mainDB: 'taxonomy',
        mainAccession: taxId.toString(),
      })}`,
  );

class Lineage extends PureComponent {
  static propTypes = {
    lineage: T.string.isRequired,
  };

  render() {
    const lineage = this.props.lineage.trim().split(' ');
    return (
      <div>
        Lineage:
        {lineage.map(taxId => {
          const LineageItemWithData = loadData(getUrlFor(taxId))(LineageItem);
          return <LineageItemWithData taxId={taxId} key={taxId} />;
        })}
      </div>
    );
  }
}

export default Lineage;
