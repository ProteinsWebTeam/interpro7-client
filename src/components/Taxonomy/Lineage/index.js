import React, { PureComponent } from 'react';
import T from 'prop-types';

import TaxIdOrName from 'components/Taxonomy/TaxIdOrName';

class Lineage extends PureComponent {
  static propTypes = {
    lineage: T.string.isRequired,
    names: T.object,
  };

  render() {
    const lineage = this.props.lineage.trim().split(' ');
    const names = this.props.names;
    return (
      <div>
        Lineage:
        {lineage.map(taxId => (
          <span key={taxId}>
            {' â†’ '}
            <TaxIdOrName accession={taxId} name={names[taxId]} />
          </span>
        ))}
      </div>
    );
  }
}

export default Lineage;
