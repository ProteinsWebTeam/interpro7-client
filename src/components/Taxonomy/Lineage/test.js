import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Lineage from '.';

const renderer = new ShallowRenderer();

describe('<Lineage />', () => {
  test('should render', () => {
    renderer.render(
      <Lineage
        lineage={' 1 131567 2 1224 28211 356 335928 99 100 '}
        names={{
          1: {
            name: 'root',
            short: 'root',
          },
          2: {
            name: 'Bacteria',
            short: 'Bacteria (eubacteria)',
          },
          99: {
            name: 'Ancylobacter',
            short: 'Ancylobacter',
          },
          100: {
            name: 'Ancylobacter aquaticus',
            short: 'Ancylobacter aquaticus',
          },
          356: {
            name: 'Rhizobiales',
            short: 'Rhizobiales (rhizobacteria)',
          },
          1224: {
            name: 'Proteobacteria',
            short: 'Proteobacteria',
          },
          28211: {
            name: 'Alphaproteobacteria',
            short: 'Alphaproteobacteria',
          },
          131567: {
            name: 'cellular organisms',
            short: 'cellular organisms',
          },
          335928: {
            name: 'Xanthobacteraceae',
            short: 'Xanthobacteraceae',
          },
        }}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
