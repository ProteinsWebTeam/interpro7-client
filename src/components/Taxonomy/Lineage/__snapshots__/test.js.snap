// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lineage /> should render 1`] = `
<div>
  <span
    className="icon icon-common primary"
    data-icon=""
  >
     
    <TaxIdOrName
      accession="1"
      name={
        {
          "name": "root",
          "short": "root",
        }
      }
    />
     
  </span>
  <span
    className="icon icon-common primary"
    data-icon=""
  >
     
    <TaxIdOrName
      accession="131567"
      name={
        {
          "name": "cellular organisms",
          "short": "cellular organisms",
        }
      }
    />
     
  </span>
  <span
    className="icon icon-common primary"
    data-icon=""
  >
     
    <TaxIdOrName
      accession="2"
      name={
        {
          "name": "Bacteria",
          "short": "Bacteria (eubacteria)",
        }
      }
    />
     
  </span>
  <span
    className="icon icon-common primary"
    data-icon=""
  >
     
    <TaxIdOrName
      accession="1224"
      name={
        {
          "name": "Proteobacteria",
          "short": "Proteobacteria",
        }
      }
    />
     
  </span>
  <span
    className="icon icon-common primary"
    data-icon=""
  >
     
    <TaxIdOrName
      accession="28211"
      name={
        {
          "name": "Alphaproteobacteria",
          "short": "Alphaproteobacteria",
        }
      }
    />
     
  </span>
  <span
    className="icon icon-common primary"
    data-icon=""
  >
     
    <TaxIdOrName
      accession="356"
      name={
        {
          "name": "Rhizobiales",
          "short": "Rhizobiales (rhizobacteria)",
        }
      }
    />
     
  </span>
  <span
    className="icon icon-common primary"
    data-icon=""
  >
     
    <TaxIdOrName
      accession="335928"
      name={
        {
          "name": "Xanthobacteraceae",
          "short": "Xanthobacteraceae",
        }
      }
    />
     
  </span>
  <span
    className="icon icon-common primary"
    data-icon=""
  >
     
    <TaxIdOrName
      accession="99"
      name={
        {
          "name": "Ancylobacter",
          "short": "Ancylobacter",
        }
      }
    />
     
  </span>
  <span
    className="icon icon-common primary"
    data-icon=""
  >
     
    <TaxIdOrName
      accession="100"
      name={
        {
          "name": "Ancylobacter aquaticus",
          "short": "Ancylobacter aquaticus",
        }
      }
    />
     
  </span>
</div>
`;
